### PR TITLE
xss attempts now filtered from Trending Communities section

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #7386 

## Description of Changes
-Added a `.filter()` to look for '"', '<', or '>' in both the first and second letters in a community name. [XSS/Cross Site Scripting](https://brightsec.com/blog/xss-attack/) usually look something like `"><script>alert()</script>` hence the filtering on the first and second letters of the name.

## Test Plan
-go to your dashboard and check to see if there are any communities that have those characters as name[0] or name[1]

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx [hunk 1]
```diff
@@ -10,6 +10,12 @@ export const TrendingCommunitiesPreview = () => {
 
   const sortedCommunities = app.config.chains
     .getAll()
+    .filter((community) => {
+      const name = community.name.toLowerCase();
+      return (
+        !['"', '>', '<'].includes(name[0]) && !['"', '>', '<'].includes(name[1])
+      );
+    })
```
Comment (burtonator) [inline packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:18]: This is awesome but we're only doing this on the frontend. I assume we're doing this on the backend as well?

Comment (burtonator) [reply packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:18]: If we are then just resolve.

Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:18]: @burtonator Yes, this was simply for the frontend to not show any of those communities within the Trending Communities column. I'll make a new ticket to fix this on the backend. Thanks for the suggestion.

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx [hunk 2]
```diff
@@ -10,6 +10,12 @@ export const TrendingCommunitiesPreview = () => {
 
   const sortedCommunities = app.config.chains
     .getAll()
+    .filter((community) => {
+      const name = community.name.toLowerCase();
+      return (
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:15]: please add the comment why we do this as without the context it looks odd

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx (hunk 2 above)
Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:17]: @masvelio Just added a comment. Good call.

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx [hunk 3]
```diff
@@ -10,6 +10,14 @@ export const TrendingCommunitiesPreview = () => {
 
   const sortedCommunities = app.config.chains
     .getAll()
+    .filter((community) => {
+      const name = community.name.toLowerCase();
+      //this filter is meant to not include any de facto communities that are actually xss attempts.
+      //It's a way of keeping the front facing parts of the app clean looking for users
+      return (
+        !['"', '>', '<'].includes(name[0]) && !['"', '>', '<'].includes(name[1])
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx:18]: Unsure, but should we also check for `/` and `'`?

Comment (Israellund) [reply packages/commonwealth/client/scripts/views/pages/user_dashboard/TrendingCommunitiesPreview/TrendingCommunitiesPreview.tsx]: @mzparacha Couldn't hurt. Double checked to see if any existing communities had either of those as the first or second characters and there were none. I added a check for the backtick to be filtered as well.

**Review by mzparacha — APPROVED** [review]

**Pull request merged**